common/persistence/history_manager.go

```diff
@@ -372,14 +372,14 @@ func ToHistoryTreeInfo(serializer serialization.Serializer, blob *commonpb.DataB
 func (m *executionManagerImpl) serializeAppendHistoryNodesRequest(
 	ctx context.Context,
 	request *AppendHistoryNodesRequest,
-) (*InternalAppendHistoryNodesRequest, error) {
+) (*InternalInsertHistoryTreeRequest, *InternalAppendHistoryNodesRequest, error) {
```

Comment:
Could you extract this to a separate method? I don't think the 3 return values look good here